libyul/SideEffects.h

```diff
@@ -30,6 +31,20 @@ namespace solidity::yul
  */
 struct SideEffects
 {
+	/// Corresponds to the effect that a YUL-builtin has on a generic data location (storage, memory
+	/// and other blockchain state).
+	enum Effect
+	{
+		None,
+		Read,
+		Write
+	};
+
+	friend Effect operator+(Effect const& _a, Effect const& _b)
+	{
+		return static_cast<Effect>(std::max(static_cast<int>(_a), static_cast<int>(_b)));
```

Comment:
```suggestion
		return static_cast<Effect>(std::max(static_cast<unsigned>(_a), static_cast<unsigned>(_b)));
```

Reply:
Why unsigned?

Reply:
Because it should never be signed (I think?)

Reply:
I used the linked example from [CppCoreGuidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#enum4-define-operations-on-enumerations-for-safe-and-simple-use) as a reference. 

I wasn't sure what would be the underlying type of the enum `Effect` since it's not specified. According to [cpp reference](https://en.cppreference.com/w/cpp/language/enum)

> Values of unscoped enumeration type are implicitly-convertible to integral types. If the underlying type is not fixed, the value is convertible to the first type from the following list able to hold their entire value range: int, unsigned int, long, unsigned long, long long, or unsigned long long. If the underlying type is fixed, the values can be converted to their promoted underlying type. 

So `int` should be the choice here, even though `unsigned` would work too.

Reply:
Ok!